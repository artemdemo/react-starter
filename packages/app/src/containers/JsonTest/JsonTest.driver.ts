import _ from 'lodash';
import { RenderResults } from '../../__tests__/testRender';

export class JsonTestDriver {
  constructor(public readonly component: RenderResults) {}

  getJson(): object {
    const { getByTestId } = this.component;
    const el = getByTestId('json-test');
    if (_.isString(el.textContent)) {
      return JSON.parse(el.textContent as string);
    }
    throw new Error("element doesn't have content");
  }
}
