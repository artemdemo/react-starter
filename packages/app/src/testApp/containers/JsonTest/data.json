{
  "glossary": {
    "title": "example glossary",
    "GlossEntry": {
      "ID": "SGML",
      "SortAs": "SGML",
      "GlossTerm": "Standard Generalized Markup Language"
    }
  }
}
