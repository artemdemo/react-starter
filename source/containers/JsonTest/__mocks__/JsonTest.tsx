import React from 'react';

const JsonTest = () => (
    <div data-mock='JsonTest' />
);

export default JsonTest;
