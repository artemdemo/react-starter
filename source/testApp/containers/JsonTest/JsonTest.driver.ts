import {RenderResults} from '../../__tests__/testRender';
import _ from 'lodash';

export class JsonTestDriver {
  constructor(public readonly component: RenderResults) {}

  getJson(): object {
    const { getByTestId } = this.component;
    const el = getByTestId('json-test');
    if (_.isString(el.textContent)) {
      return JSON.parse(el.textContent as string);
    }
    throw new Error('element doesn\'t have content');
  }
}
