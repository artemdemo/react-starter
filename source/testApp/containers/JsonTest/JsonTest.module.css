.JsonTest {
  font-size: 0.8em;
  background: #f4f6f7;
  border: 1px solid #ECEFF1;
  padding: 5px 8px;
  border-radius: 3px;
  display: inline-block;
}
