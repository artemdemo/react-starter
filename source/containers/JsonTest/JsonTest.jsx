import React from 'react';
import data from './data.json';

const JsonTest = () => {
    return (
        <React.Fragment>
            <p>
                JSON import example:
            </p>
            <pre>
                {JSON.stringify(data, null, 2)}
            </pre>
        </React.Fragment>
    );
};

export default JsonTest;
