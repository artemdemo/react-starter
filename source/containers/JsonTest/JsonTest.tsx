import React from 'react';
import styled from 'styled-components';
import data from './data.json';

/*
 * There have been problems with importing json for certain versions of webpack.
 * Therefore I'm testing it to be sure that I can update to the latest version.
 */

const JsonContent = styled.pre`
    font-size: 0.8em;
    background: #f4f6f7;
    border: 1px solid #ECEFF1;
    padding: 5px 8px;
    border-radius: 3px;
    display: inline-block;
`;

const JsonTest = () => {
    return (
        <>
            <p>
                JSON import example:
            </p>
            <JsonContent>
                {JSON.stringify(data, null, 2)}
            </JsonContent>
        </>
    );
};

export default JsonTest;
