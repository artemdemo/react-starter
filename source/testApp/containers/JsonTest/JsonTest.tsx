import React from 'react';
import s from './JsonTest.module.css';
import data from './data.json';

/*
 * There have been problems with importing json for certain versions of webpack.
 * Therefore I'm testing it to be sure that I can update to the latest version.
 */

export const JsonTest = () => {
  return (
    <>
      <p>JSON import example:</p>
      <pre className={s.JsonTest} data-testid="json-test">{JSON.stringify(data, null, 2)}</pre>
    </>
  );
};
