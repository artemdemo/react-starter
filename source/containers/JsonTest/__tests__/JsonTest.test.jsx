import React from 'react';
import renderer from 'react-test-renderer';
import JsonTest from '../JsonTest';

describe('<JsonTest>', () => {
    it('Empty render', () => {
        const tree = renderer.create(
            <JsonTest />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
