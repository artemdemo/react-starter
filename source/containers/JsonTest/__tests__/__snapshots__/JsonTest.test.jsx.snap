// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JsonTest> Empty render 1`] = `
Array [
  <p>
    JSON import example:
  </p>,
  <pre
    className="sc-AxjAm bCeuRk"
  >
    {
  "glossary": {
    "title": "example glossary",
    "GlossEntry": {
      "ID": "SGML",
      "SortAs": "SGML",
      "GlossTerm": "Standard Generalized Markup Language"
    }
  }
}
  </pre>,
]
`;
